### DISCUSSION TITLE
can I put a default response_model when including a router?

### DISCUSSION THREAD

**Discussion by joaosvictor** [fastapi/fastapi#7378] [Questions] [answered]
### First checkHow can I put a default response_model when including a router? For example, having this router:

# routers/users.py

router = APIRouter()

@router.get('/{user_id}', response_model=User)
def get_user(user_id: int, db: Session = Depends(get_db)):
    return crud.get_user(db=db, user_id=user_id)

@router.put('/{user_id}', response_model=User)
def update_user(user_id: int, user: UserUpdate, db: Session = Depends(get_db)):
    return crud.update_user(db=db, user_id=user_id, user=user)
I am using the same response_model in all routes, so I might rather put it when including the router:

# main.py

app.include_router(users.router, prefix='/users', default_response_model=User)
But this gives an error: TypeError: include_router() got an unexpected keyword argument 'default_response_model' for both default_response_model and response_model.

Is there any reason why this is not supported?

* [ ] I added a very descriptive title to this issue.
* [ ] I used the GitHub search to find a similar issue and didn't find it.
* [ ] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [ ] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}
```

### Description

* Open the browser and call the endpoint `/`.
* It returns a JSON with `{"Hello": "World"}`.
* But I expected it to return `{"Hello": "Sara"}`.

### Environment

* OS: [e.g. Linux / Windows / macOS]:
* FastAPI Version [e.g. 0.3.0]:

To know the FastAPI version use:

```bash
python -c "import fastapi; print(fastapi.__version__)"
```

* Python version:

To know the Python version use:

```bash
python --version
```

**Comment by Kludex** [comment]
@joaovictorsantos If you want people to help you, at least try to make it easy... 

> can I put a default response_model when including a router?

No.

```
app.include_router(users.router, prefix='/users', default_response_model=User)
``` 
> But this gives an error: TypeError: include_router() got an unexpected keyword argument 'default_response_model' for both default_response_model and response_model.
Is there any reason why this is not supported?

My opinion: I don't see the usefulness of having this feature... But if there is a good reason for having it, it will be nice if you could share it. :thinking:

**Comment by joaosvictor** [comment]
if u not bad it, i could share u.

**Comment by Kludex** [comment, marked as answer]
Man, I'm going to recommend you to go to the FastAPI Brazilian Telegram group, where you don't need to face the language obstacle: https://t.me/fastapibr

Mano, vou te recomendar o grupo do FastAPI Brasil do telegram. Lá você vai poder perguntar o que quiser em português: https://t.me/fastapibr

**Comment by joaosvictor** [comment]
Okay,  I appreciate that! I'll join in future as possible

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex! 🍰 

And thanks for coming back to close the issue @joaosvictor ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
